test/sol5/kyberNetworkProxy.js

```diff
@@ -578,6 +602,392 @@ contract('KyberNetworkProxy', function(accounts) {
+                srcQty,
+                tokens[2].address,
+                taker,
+                maxDestAmt,
+                rate.expectedRate,
+                zeroAddress,
+                { from: taker }
+            );
+        });
+
+        it("trade revert if actual dest amount is different from reported", async () => {
+            await networkProxy.setKyberNetwork(generousNetwork.address, { from: admin });
+            let srcToken = tokens[1];
+            let dstToken = tokens[2]
+            let amountTwei = 1515;
+
+            tokenAmountForTrades = new BN(10000).mul(new BN(10).pow(tokenDecimals[1]));
+            await srcToken.transfer(taker, amountTwei);
+            await dstToken.transfer(generousNetwork.address, tokenAmountForTrades);
+            //get rate
+            let rate = await networkProxy.getExpectedRate(srcToken.address, dstToken.address, amountTwei);
+            await srcToken.approve(networkProxy.address, amountTwei, { from: taker });
+
+            //see trade reverts
+            await expectRevert(
+                networkProxy.trade(
+                    srcToken.address,
+                    amountTwei,
+                    dstToken.address,
+                    taker,
+                    maxDestAmt,
+                    rate.expectedRate,
+                    zeroAddress,
+                    { from: taker }
+                ), "wrong ret amount"
+            );
+            // change the amount and see trade success
+            await networkProxy.trade(
+                srcToken.address,
+                1514,
+                tokens[2].address,
+                taker,
+                maxDestAmt,
+                rate.worstRate,
+                zeroAddress,
+                { from: taker }
+            );
+        });
+
+        it("trade revert if src balance after is greater or equal to balance before", async () => {
+            await networkProxy.setKyberNetwork(generousNetwork.address, { from: admin });
+            let srcToken = tokens[1];
+            let dstToken = tokens[2];
+            let amountTwei = 1313;
+
+            tokenAmountForTrades = new BN(10000).mul(new BN(10).pow(tokenDecimals[1]));
+            await srcToken.transfer(taker, amountTwei);
+            await srcToken.transfer(generousNetwork.address, tokenAmountForTrades);
+            //get rate
+            let rate = await networkProxy.getExpectedRate(srcToken.address, dstToken.address, amountTwei);
+            await srcToken.approve(networkProxy.address, amountTwei, { from: taker });
+
+            //see trade reverts
+            await expectRevert(
+                networkProxy.trade(
+                    srcToken.address,
+                    amountTwei,
+                    dstToken.address,
+                    taker,
+                    maxDestAmt,
+                    rate.expectedRate,
+                    zeroAddress,
+                    { from: taker }
+                ), "srcAdd bad qty"
```

Comment:
please update this message in proxy contract to 
Wrong amount in source address.
And change here as well.

Reply:
updated in https://github.com/KyberNetwork/smart-contracts/pull/726/commits/e4d067f71b9400d43a678940a586bbb43991f02a